python/ray/tune/schedulers/hyperband.py

```diff
@@ -276,7 +276,7 @@ def _process_bracket(
             # kill bad trials
             self._num_stopped += len(bad)
             for t in bad:
-                if t.status == Trial.PAUSED:
+                if t.status == Trial.PAUSED or t.is_saving:
```

Comment:
May lose the checkpoint here if we stop while saving?

Reply:
This change is avoiding the situation when the trial is in the middle of pausing:

```
            if should_checkpoint:
                self._cached_trial_decisions[trial.trial_id] = TrialScheduler.PAUSE
                future_result = self._schedule_trial_save(
                    trial=trial, storage=CheckpointStorage.PERSISTENT
                )
                trial.temporary_state.saving_to = future_result
```

where the status is technically still "RUNNING" while the save is resolving.

I think it's fine to miss the checkpoint here. The trial is bad anyway and is going to be early-terminated.